### ISSUE TITLE
Use hardhat verify instead of hardhat etherscan-verify on yarn verify

### ISSUE THREAD

**Issue by damianmarti** [scaffold-eth/scaffold-eth-2#314]
**hardhat verify** has more features, like linking to libraries.

**Comment by metoinside** [comment]
Hello mates,

#### TLDR: 
I didn't know `hardhat-etherscan` [plugin](https://github.com/NomicFoundation/hardhat/tree/main/packages/hardhat-etherscan) depreciated, so I'm prooving below that why we need to move `hardhat-verify`, lol 😄 

---

We can `yarn verify` successfully on _polygonMumbai_, _sepolia_ etc. 
However, we're unable to do the same in _polygonZkEvmTestnet_, _arbitriumNova_, _scrollSepolia_ etc. networks.  
This is something outside of the [API keys issue](https://github.com/scaffold-eth/scaffold-eth-2/issues/498) as you may notice the error message is different.
![telegram-cloud-photo-size-4-5976723026093391279-y](https://github.com/scaffold-eth/scaffold-eth-2/assets/5622253/b8e2ca09-2530-43d1-b8c5-e30a191c4360)



---

### What might be the reason?
I learned that **hardhat-etherscan** supports the below list.
After reviewing it you'll better understand why the same contract verification 
- ✅ works on _arbitriumGoerli_ 
- 🚫 but doesn't work for _arbitrumSepolia_ and, others such as:
  - _polygonZkEvm_, _polygonZkEvmTestnet_, _zkSync_, _zkSyncTestnet_, [_base_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7), [_baseGoerli_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7), [_scroll_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7), [_scrollSepolia_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7)
    - they don't overlap with the list of hardhat-etherscan therefore it returns the error

<img width="561" alt="image" src="https://github.com/scaffold-eth/scaffold-eth-2/assets/5622253/b6dae73c-6b91-4cc4-be39-a54a13126141">

----

Anyway, if you are busy, I'm happy to take over this task to complete on my free time, @damianmarti and @austintgriffith

**Comment by damianmarti** [comment]
> Hello mates,
> 
> #### TLDR:
> I didn't know `hardhat-etherscan` [plugin](https://github.com/NomicFoundation/hardhat/tree/main/packages/hardhat-etherscan) depreciated, so I'm prooving below that why we need to move `hardhat-verify`, lol 😄
> 
> We can `yarn verify` successfully on _polygonMumbai_, _sepolia_ etc. However, we're unable to do the same in _polygonZkEvmTestnet_, _arbitriumNova_, _scrollSepolia_ etc. networks. This is something outside of the [API keys issue](https://github.com/scaffold-eth/scaffold-eth-2/issues/498) as you may notice the error message is different. ![telegram-cloud-photo-size-4-5976723026093391279-y](https://private-user-images.githubusercontent.com/5622253/285594879-b8e2ca09-2530-43d1-b8c5-e30a191c4360.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwOTgwNTUsIm5iZiI6MTcwMTA5Nzc1NSwicGF0aCI6Ii81NjIyMjUzLzI4NTU5NDg3OS1iOGUyY2EwOS0yNTMwLTQzZDEtYjhjNS1lMzBhMTkxYzQzNjAuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjdUMTUwOTE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGQ4NGRkYmQxMGJiMDdmMGI4NmRmYWM5MWVlMzNjNzllN2JiZWM2Y2EyOTBkYTNiNGJhODMwYTk3MjAzMDBlOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6K8zG1RiQFeNaWeMz3j_MG8GLrzQuV_TfsF70LV2TkI)
> 
> ### What might be the reason?
> I learned that **hardhat-etherscan** supports the below list. After reviewing it you'll better understand why the same contract verification
> 
> * ✅ works on _arbitriumGoerli_
> * 🚫 but doesn't work for _arbitrumSepolia_ and, others such as:
>   
>   * _polygonZkEvm_, _polygonZkEvmTestnet_, _zkSync_, _zkSyncTestnet_, [_base_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7), [_baseGoerli_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7), [_scroll_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7), [_scrollSepolia_](https://github.com/scaffold-eth/scaffold-eth-2/blob/55ae5a4333bc70d1fb6e73eac11850914d7b62de/packages/hardhat/hardhat.config.ts#L109C1-L124C7)
>     
>     * they don't overlap with the list of hardhat-etherscan therefore it returns the error
> 
> <img alt="image" width="561" src="https://private-user-images.githubusercontent.com/5622253/285607668-b6dae73c-6b91-4cc4-be39-a54a13126141.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwOTgwNTUsIm5iZiI6MTcwMTA5Nzc1NSwicGF0aCI6Ii81NjIyMjUzLzI4NTYwNzY2OC1iNmRhZTczYy02YjkxLTRjYzQtYmUzOS1hNTRhMTMxMjYxNDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjdUMTUwOTE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGNlYjdiMjk1ZDg5N2RiMTA0MGIxMjczYWQzYTFhOWFlM2VkNjg2Zjc4MjA3NDg2YWQ0ZTc0MGZhOGFkZTAzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.16rVHr2VGKKx5wCTSR3OLwfs1okXhdjP241hAJlcels">
> Anyway, if you are busy, I'm happy to take over this task to complete on my free time, @damianmarti and @austintgriffith

Yes, sure! Thanks!!

**Comment by metoinside** [comment]
Today, I was able to verify two contracts on _scrollSepolia_ and _polygonZkEvmTestnet_ as an example with `hardhat-verify`. I learned quite a lot and below are the links of the contracts: 
- [sepolia.scrollscan.com/address/0xfdd6...](https://sepolia.scrollscan.com/address/0xfdd6c1a247ab7b4ea06fb9dc3fc00ea724a6131c#code)
- [testnet-zkevm.polygonscan.com/address/0xdd85...](https://testnet-zkevm.polygonscan.com/address/0xdd85aca5dc954f0dffe5575ade47bc193fd18bfe#code)

However, I noticed it requires quite a lot to update in existing libraries in order to use `hardhat-verify` therefore, I'll create the PR after having some housekeeping
